### PR TITLE
[S6-2112] Save and display previous test runs

### PR DESCRIPTION
**Pull request by bee-anchor**

### CODE REVIEW

File: bfgg-site/src/components/AgentStatus/index.jsx [hunk 1]
```diff
@@ -44,7 +43,7 @@ function EnhancedTableHead(props) {
                 indeterminate={numSelected > 0 && numSelected < rowCount}
                 checked={numSelected === rowCount}
                 onChange={onSelectAllClick}
-                inputProps={{ 'aria-label': 'select all desserts' }}
+                inputProps={{ 'aria-label': 'select all agents' }}
```
Comment (David-IW) [inline bfgg-site/src/components/AgentStatus/index.jsx]: goodbye desserts :(

File: bfgg/controller/actors/dynamodb_resource.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            'TestId': id,
+            'StartTime': start_time.isoformat(),
+            'Project': project,
+            'TestClass': test_class,
+        }
+        if java_opts:
+            item['JavaOpts'] = java_opts
+        self.table.put_item(Item=item)
+
+    def update_test_ended(self, id: str, end_time: datetime, test_results_url: str):
```
Comment (David-IW) [inline bfgg/controller/actors/dynamodb_resource.py]: this works fine, but its only called if a test successfully finishes - if theres an error or test is manually stopped the end date isnt updated, is this intentional?

File: bfgg-site/src/components/PastTestsTable/index.jsx [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+}));
+
+export default function PastTestsTable(props) {
+  const {selectedGroup, setSnackbar, setSnackbarOpen } = props;
+  const classes = useStyles();
+  const [order, setOrder] = React.useState('asc');
+  const [orderBy, setOrderBy] = React.useState('StartTime');
+  const [ tests, setTests ] = useState([]);
+
+  const handleRequestSort = (event, property) => {
```
Comment (David-IW) [inline bfgg-site/src/components/PastTestsTable/index.jsx]: seems like this function doesnt work properly, the sorting isnt updating when clicking different column headers

File: bfgg/controller/actors/dynamodb_resource.py (hunk 2 above)
Comment (bee-anchor) [reply bfgg/controller/actors/dynamodb_resource.py]: yeah, you wont get the results and end date if there was an issue with the test, but the details of how the test was run are still available to you. Allows you to rerun them even if they didn't finish, and gives a way to differentiate between 'run' and 'run & completed' tests

File: bfgg-site/src/components/PastTestsTable/index.jsx (hunk 3 above)
Comment (bee-anchor) [reply bfgg-site/src/components/PastTestsTable/index.jsx]: 😭

File: bfgg-site/src/components/AgentStatus/index.jsx (hunk 1 above)
Comment (bee-anchor) [reply bfgg-site/src/components/AgentStatus/index.jsx]: 🍰

**Pull request merged**